lib/jsonapi/utils/string.ex

```diff
@@ -111,10 +114,18 @@ defmodule JSONAPI.Utils.String do
              ~r{(?<=[a-zA-Z0-9])[-_](?=[a-zA-Z0-9])},
              to_string(value)
            ) do
-      [h | t] = words |> Enum.filter(&(&1 != ""))
-
-      [String.downcase(h) | camelize_list(t)]
-      |> Enum.join()
+      tokens = Enum.filter(words, &(&1 != ""))
+
+      case tokens do
+        # If there is only one token, leave it as-is
+        [token] ->
+          token
```

Comment:
Annotation: This idea here is that if we could not produce a list of multiple tokens, i.e. there was no `-` or `_` delimiting characters, to return the single token as-is.